src/Compilers/CSharp/Portable/Lowering/StateMachineRewriter/SynthesizedStateMachineMethod.cs

```diff
@@ -9,19 +10,18 @@ namespace Microsoft.CodeAnalysis.CSharp
     /// <summary>
     /// State machine interface method implementation.
     /// </summary>
-    internal sealed class SynthesizedStateMachineMethod : SynthesizedImplementationMethod, ISynthesizedMethodBodyImplementationSymbol
+    internal abstract class SynthesizedStateMachineMethodBase : SynthesizedImplementationMethod, ISynthesizedMethodBodyImplementationSymbol
```

Comment:
This is a poor name; it represents all synthesized state machines, so it should be called `SynthesizedStateMachineMethod`. The more specialized subclasses can have more specific names, but we should avoid having the code's behavior based on the shape of the hierarchy of implementation classes.